test/numerical/TestMatMul2D.cpp

```diff
@@ -23,8 +23,21 @@ static bool isOMMatmulTheSameAsNaiveImplFor(
   static int testNum = 0;
   printf("attempt %d with i %d, j %d, k %d\n", ++testNum, I, J, K);
   MatMul2DLibBuilder matmul(SHARED_LIB_BASE.str(), I, J, K);
-  return matmul.build() && matmul.compileAndLoad() && matmul.prepareInputs() &&
-         matmul.run() && matmul.verifyOutputs();
+  bool successBuild = matmul.build() && matmul.compileAndLoad();
+  if (!successBuild) {
+    printf("Fail to build.\n");
+    return false;
+  }
+  // Verify generated library
+  std::string instructionName =
+      getenv("TEST_CHECK_INSTRUCTION") ? getenv("TEST_CHECK_INSTRUCTION") : "";
+  std::string sharedLibName =
+      ModelLibBuilder::getSharedLibName(SHARED_LIB_BASE.str());
+  if (!ModelLibBuilder::checkSharedLibInstruction(
+          instructionName, sharedLibName))
+    return false;
+  return successBuild && matmul.prepareInputs() && matmul.run() &&
+         matmul.verifyOutputs();
```

Comment:
same here

Reply:
Thanks for your comments. Refactoring done.